pkg/controller/infrastructure/templates/main.tpl.tf

```diff
@@ -144,19 +163,33 @@ resource "aws_security_group_rule" "nodes_egress_all" {
   from_port         = 0
   to_port           = 0
   protocol          = "-1"
+  {{ if .isIPv4 }}
   cidr_blocks       = ["0.0.0.0/0"]
+  {{ end }}
+  {{ if .isIPv6 }}
+  ipv6_cidr_blocks  = ["::/0"]
+  {{ end }}
   security_group_id = aws_security_group.nodes.id
 }
 
 {{ range $index, $zone := .zones }}
 resource "aws_subnet" "nodes_z{{ $index }}" {
   vpc_id            = {{ $.vpc.id }}
+  {{ if $.isIPv4 }}
   cidr_block        = "{{ $zone.worker }}"
+  {{ end }}
   availability_zone = "{{ $zone.name }}"
-{{- if $.dualStack.enabled }}
+
+  {{ if $.isIPv6 }}
+  ipv6_native = true
+  assign_ipv6_address_on_creation = true
+  ipv6_cidr_block = "${cidrsubnet({{ $.vpc.ipv6CidrBlock }}, 8, (({{ $index }} * 3)))}"
+  enable_resource_name_dns_aaaa_record_on_launch = true
+  {{ end }}
+  {{- if $.dualStack.enabled }}
   ipv6_cidr_block = "${cidrsubnet({{ $.vpc.ipv6CidrBlock }}, 8, (({{ $index }} * 3)))}"
   assign_ipv6_address_on_creation = false
```

Comment:
don't this settings collide with the IPv6 settings ?

Reply:
Added the missing `else`.